app/src/main/java/org/dhis2/usescases/notes/NotesFragment.kt

```diff
@@ -116,15 +116,21 @@ class NotesFragment : FragmentGlobalAbstract(), NotesView {
         super.onPause()
     }
 
-    override fun swapNotes(noteModules: List<Note>) {
-        noteAdapter.setItems(noteModules)
+    override fun swapNotes(notes: List<Note>) {
+        if (notes.isEmpty()) {
+            binding.noNotesLayout.visibility = View.VISIBLE
+            binding.notesRecycler.visibility = View.GONE
+        } else {
+            binding.noNotesLayout.visibility = View.GONE
+            binding.notesRecycler.visibility = View.VISIBLE
+            noteAdapter.setItems(notes)
+        }
```

Comment:
I think this logic could go in the presenter and call methods view.showNotes or not . It's small but doing this we put less responsability in view and can be unit test.